domino/api/src/main/java/io/quarkus/domino/ProjectDependencyResolver.java

```diff
@@ -752,7 +763,7 @@ private void processRootArtifact(ArtifactCoords rootArtifact) {
             }
             for (DependencyNode d : root.getChildren()) {
                 if (d.getDependency().isOptional()
-                        && !(config.isIncludeOptionalDeps() || isIncluded(toCoords(d.getArtifact())))) {
+                        && !(config.isIncludeOptionalDeps() || isExplicitlyIncluded(toCoords(d.getArtifact())))) {
```

Comment:
why do we only consult the include lest when deciding if we should recurse into optional deps?